### PR TITLE
Improve handling of vertical grid, adding support for MOM6 grids

### LINKED ISSUE #33: MOM6 vgrid support
These tools currently assume a MOM5 vgrid is used, which has 2n+1 values for an n-level grid. They therefore produce incorrect results if a MOM6 vgrid with n+1 levels is used.

Perhaps an option should be provided to use MOM6 vertical grids?

### PR DESCRIPTION
**Pull request by micaeljtoliveira**
This PR adds support for MOM6 vertical grids. The type of grid used is controlled via a command line option.

Closes #33

### CODE REVIEW

**Comment by micaeljtoliveira** [comment]
@aekiss Would you mind having a quick look? There's a lot of stuff going on, but would be good to know if the documentation is okay.

File: src/vgrid.f90 [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+      allocate(character(len=history_len) :: vgrid%history)
+      call handle_error(nf90_get_att(ncid, nf90_global, 'history', vgrid%history))
+    end if
+
+    ! Handle the different types of grids
+    select case (type)
+    case ("mom5")
+      if (mod(zeta_len, 2) == 0) then
+        write(error_unit,'(a)') "ERROR: MOM5 vertical grid has an even number of points, which should never happen."
+      error stop
```
Comment (aekiss) [inline src/vgrid.f90]:
```suggestion
        error stop
```

File: src/vgrid.f90 [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+    class(vgrid_t), intent(inout) :: this
+    character(len=*), intent(in) :: command
+
+    character(len=:), allocatable :: new_history, old_history
+
+    new_history = date_time() // ": " // trim(command)
+    if (allocated(this%history)) then
+      old_history = this%history
+      deallocate(this%history)
+      this%history = trim(new_history) // C_NEW_LINE // old_history
```
Comment (aekiss) [inline src/vgrid.f90:182]: reverse chronological order seems counterintuitive to me

**Review by aekiss — DISMISSED** [review]
Thanks @micaeljtoliveira, LGTM and documentation is clear

File: README.md
Comment (aekiss) [inline README.md:1]: need to update line 9

Comment (aekiss) [inline README.md:1]: add `vgrid_type` to
```
usage: topogtools fix_nonadvective --input <input_file> --output <output_file>
                                   [--vgrid <vgrid> --potholes --coastal-cells]
```
https://github.com/COSIMA/domain-tools/pull/35/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R98

File: src/vgrid.f90 (hunk 2 above)
Comment (micaeljtoliveira) [reply src/vgrid.f90:182]: IIRC, this is how other tools that update the history do it, so I just kept it like that for consistency.

**Comment by micaeljtoliveira** [comment]
@aekiss Thanks for the review! I think I fixed everything. Have a look and please approve if you think it's good to be merged.

**Review by aekiss — APPROVED** [review]
LGTM

**Pull request merged**